src/Microsoft.AspNetCore.Mvc.Razor/RazorPageActivator.cs

```diff
@@ -63,25 +63,34 @@ public void Activate(IRazorPage page, ViewContext context)
             }
 
             var pageType = page.GetType();
-            RazorPagePropertyActivator propertyActivator;
-            if (!_activationInfo.TryGetValue(pageType, out propertyActivator))
+            if (!_activationInfo.TryGetValue(pageType, out var result))
             {
                 // Look for a property named "Model". If it is non-null, we'll assume this is
                 // the equivalent of TModel Model property on RazorPage<TModel>.
                 //
                 // Otherwise if we don't have a model property the activator will just skip setting
                 // the view data.
                 var modelType = pageType.GetRuntimeProperty(ModelPropertyName)?.PropertyType;
-                propertyActivator = new RazorPagePropertyActivator(
-                    pageType,
-                    modelType,
-                    _metadataProvider,
-                    _propertyAccessors);
 
-                propertyActivator = _activationInfo.GetOrAdd(pageType, propertyActivator);
+                var viewDataDictionaryFactory = new RazorPageViewDataDictionaryFactory(_metadataProvider, modelType);
+
+                RazorPagePropertyActivator propertyActivator = null;
+                if (!pageType.IsDefined(typeof(SkipRazorPagePropertyActivationAttribute)))
```

Comment:
When a Razor Page gets executed as part of the view, we want to 

a) create a nested `ViewData` so that it picks up stuff set by a _ViewStart
b) ensure that `PageContext.ViewData` = `ViewContext.ViewData` so we don't have two different instances depending on how you access the ViewData
c) not activate any of the properties

If we use the attribute instead of `RazorPageAdapter`, we can keep the existing modelType inference code here unchanged. We want the modelType to be preserved to get the right `VDD<Model>`